src/dsp_tools/commands/ingest_xmlupload/bulk_ingest_client.py

```diff
@@ -59,29 +59,27 @@ def upload_file(
         filepath: Path,
     ) -> UploadFailure | None:
         """Uploads a file to the ingest server."""
-        try:
```

Comment:
*suggestion* Keep separate try blocks if not other reasons force you to do it.

In my experience as soon as code is raising an Error it is generally a harder thing to understand the control flow, especially when reading the code. In a `try` block with multiple lines and `except`s it is not always obvious which line could have caused the `Error`. 

I always try minimizing the number of lines in a `try` like here, the OSError can only come from the open or the read and not from something else.